src/core/layout/qgslayoutitemmap.cpp

```diff
@@ -2226,7 +2232,12 @@ void QgsLayoutItemMap::refreshMapExtents( const QgsExpressionContext *context )
   QgsExpressionContext scopedContext;
   if ( !context )
     scopedContext = createExpressionContext();
+
+  bool ok = false;
```

Comment:
Can you move this block of code from refreshMapExtents to QgsLayoutItemMap::refreshDataDefinedProperty, and follow the same approach used elsewhere in there? (i.e. refresh the crs if `property == QgsLayoutObject::MapCrs` || property == QgsLayoutObject::AllProperties`).

Put it before the block which refreshes the map extents and it should work as intended.

Reply:
I did something but I'm not sure if I need to add a signal or if that solution is what you had in mind.

Reply:
Looks perfect, thanks!